File: app/src/main/java/com/bannersimple/refresh/ArrayUtils.java
package com.bannersimple.refresh;

import android.support.annotation.NonNull;

import com.bannersimple.bean.SimpleBannerModel;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * by y on 2017/4/11
 */

public class ArrayUtils {


    public static List<SimpleBannerModel> initArrayResources(@NonNull Object[] imageArray) {
        List<SimpleBannerModel> imageArrayList = new ArrayList<>();
        for (Object url : Arrays.asList(imageArray)) {
            imageArrayList.add(new SimpleBannerModel(url));
        }
        return imageArrayList;
    }

    public static List<SimpleBannerModel> initArrayResources(
            @NonNull Object[] imageArray,
            @NonNull String[] imageArrayTitle) {
        List<Object> url = Arrays.asList(imageArray);
        List<String> title = Arrays.asList(imageArrayTitle);
        List<SimpleBannerModel> imageArrayList = new ArrayList<>();
        SimpleBannerModel bannerModel;
        for (int i = 0; i < url.size(); i++) {
            bannerModel = new SimpleBannerModel();
            bannerModel.setImage(url.get(i));
            bannerModel.setTitle(title.get(i));
            imageArrayList.add(bannerModel);
        }

        return imageArrayList;
    }

}

File: app/src/main/java/com/bannersimple/refresh/ArrayUtils.kt
package com.bannersimple.refresh

import com.bannersimple.bean.SimpleBannerModel
import java.util.*

/**
 * by y on 2017/4/11
 */

object ArrayUtils {

    fun initArrayResources(
            imageArray: Array<Any>,
            imageArrayTitle: Array<String>): MutableList<SimpleBannerModel> {
        val url = Arrays.asList(*imageArray)
        val title = Arrays.asList(*imageArrayTitle)
        val imageArrayList = ArrayList<SimpleBannerModel>()
        var bannerModel: SimpleBannerModel
        for (i in url.indices) {
            bannerModel = SimpleBannerModel(url[i], title[i])
            imageArrayList.add(bannerModel)
        }
        return imageArrayList
    }

}
